consensus/safety-rules/src/safety_rules.rs

```diff
@@ -235,6 +235,42 @@ impl<T: Payload> TSafetyRules<T> for SafetyRules<T> {
     /// @TODO verify QC matches preferred round
     fn sign_proposal(&mut self, block_data: BlockData<T>) -> Result<Block<T>, Error> {
         debug!("Incoming proposal to sign.");
+        let proposal_epoch = block_data.epoch();
+        let proposal_round = block_data.round();
+        let last_voted_round = self.persistent_storage.last_voted_round()?;
+
+        // proposed epoch must match the current epoch
+        self.verify_epoch(proposal_epoch)?;
+
+        // proposed round must be increasing
```

Comment:
same here, we have a debug statement below describing why we have this block of code, so a comment is just more overhead

Reply:
Leaving comments in the code is kind of a personal habbit. Will remove them thereafter..

Reply:
I prefer code as comments too, only leave comments when the code is not obvious